engine/runtime/src/main/java/org/enso/interpreter/node/controlflow/caseexpr/LiteralBranchNode.java

```diff
@@ -0,0 +1,52 @@
+package org.enso.interpreter.node.controlflow.caseexpr;
+
+import com.oracle.truffle.api.RootCallTarget;
+import com.oracle.truffle.api.dsl.Cached;
+import com.oracle.truffle.api.dsl.Fallback;
+import com.oracle.truffle.api.dsl.Specialization;
+import com.oracle.truffle.api.frame.VirtualFrame;
+import com.oracle.truffle.api.interop.InteropLibrary;
+import com.oracle.truffle.api.library.CachedLibrary;
+import com.oracle.truffle.api.nodes.NodeInfo;
+import com.oracle.truffle.api.profiles.ConditionProfile;
+import org.enso.interpreter.node.expression.builtin.text.util.ToJavaStringNode;
+import org.enso.interpreter.runtime.data.text.Text;
+
+@NodeInfo(shortName = "LiteralMatch", description = "Allows matching on literals")
+public abstract class LiteralBranchNode extends BranchNode {
+  private final Object literal;
+
+  private final ConditionProfile textProfile = ConditionProfile.createCountingProfile();
+  private final ConditionProfile numProfile = ConditionProfile.createCountingProfile();
+
+  LiteralBranchNode(Object literal, RootCallTarget branch) {
+    super(branch);
```

Comment:
What are the actual values of `literal`? `String`, `long`, `EnsoBigInteger`? Should they be enforced? By `assert`? By multiple variants of the constructor?

Reply:
done